sdk/src/utils/pdf_utils.rs

```diff
@@ -0,0 +1,507 @@
+        let names_dictionary = match catalog_names.as_reference() {
+            Ok(object_id) => self.document.get_object_mut(object_id)?.as_dict_mut()?,
+            _ => catalog_names.as_dict_mut()?,
+        };
+
+        let Ok(embedded_files) = names_dictionary.get_mut(EMBEDDED_FILES_KEY) else {
+            // We have a /Names dictionary, but are missing the /EmbeddedFiles dictionary
+            // and its /Names array of embedded files.
+            names_dictionary.set(
+                EMBEDDED_FILES_KEY,
+                dictionary! { NAMES_KEY => manifest_name_file_pair },
+            );
+            return Ok(());
+        };
+
+        // Follows the reference to the /EmbeddedFiles Dictionary, if the Object is a Reference.
+        let embedded_files_dictionary = match embedded_files.as_reference() {
+            Ok(object_id) => self.document.get_object_mut(object_id)?.as_dict_mut()?,
+            _ => embedded_files.as_dict_mut()?,
+        };
+
+        let Ok(names) = embedded_files_dictionary.get_mut(NAMES_KEY) else {
+            // This PDF has the /Names dictionary, and it has the /EmbeddedFiles
+            // dictionary, but the /EmbeddedFiles Dictionary is missing the /Names Array.
+            embedded_files_dictionary.set(
+                NAMES_KEY,
+                dictionary! { NAMES_KEY => manifest_name_file_pair },
+            );
+
+            return Ok(());
+        };
+
+        // Follows the reference to the /Names Array, if the Object is a Reference.
+        let names_array = match names.as_reference() {
+            Ok(object_id) => self.document.get_object_mut(object_id)?.as_array_mut()?,
+            _ => names.as_array_mut()?,
+        };
+
+        // The PDF has the /Names dictionary, which contains the /EmbeddedFiles Dictionary, which
+        // contains the /Names array. Append the manifest's name (Content Credentials)
+        // and its reference.
+        names_array.append(&mut manifest_name_file_pair);
+
+        Ok(())
+    }
+
+    /// Writes the provided bytes to the PDF as a `FileAttachment` `Annotation`. This `Annotation`
+    /// is added to the first page of the `PDF`, to the lower left corner.
+    fn write_manifest_as_annotation(&mut self, bytes: Vec<u8>) -> Result<(), Error> {
+        let file_stream_reference = self.add_c2pa_embedded_file_stream(bytes);
+        let file_spec_reference = self.add_embedded_file_specification(file_stream_reference);
+
+        self.set_af_relationship(file_spec_reference)?;
+        self.add_file_attachment_annotation(file_spec_reference)?;
+
+        Ok(())
+    }
+
+    /// Gets a reference to the `C2PA` manifest bytes of the PDF.
+    ///
+    /// This method will read the bytes of the manifest, whether the manifest was added to the
+    /// PDF via an `Annotation` or an `EmbeddedFile`.
+    ///
+    /// Returns an [Ok(None)] if no manifest is present. Returns a [Ok(Some(&Vec<u8>))] when a manifest
```

Comment:
I think these values should be wrapped with backticks and not braces for proper Markdown formatting.